pkg/serializer/split/split.go

```diff
@@ -56,24 +57,24 @@ func init() {
 
 // CheckSizeAndSerialize Check the size of a payload and marshall it (optionally compress it)
 // The dual role makes sense as you will never serialize without checking the size of the payload
-func CheckSizeAndSerialize(m marshaler.Marshaler, compress bool, mType MarshalType) (bool, []byte, []byte, error) {
+func CheckSizeAndSerialize(m marshaler.Marshaler, compress bool, mType MarshalType) (bool, bool, []byte, []byte, error) {
 	compressedPayload, payload, err := serializeMarshaller(m, compress, mType)
 	if err != nil {
-		return false, nil, nil, err
+		return false, false, nil, nil, err
 	}
-	return checkSize(compressedPayload), compressedPayload, payload, nil
+	return checkSizeCompressed(compressedPayload), checkSizeUnCompressed(payload), compressedPayload, payload, nil
 }
 
 // Payloads serializes a metadata payload and sends it to the forwarder
 func Payloads(m marshaler.Marshaler, compress bool, mType MarshalType) (forwarder.Payloads, error) {
 	marshallers := []marshaler.Marshaler{m}
 	smallEnoughPayloads := forwarder.Payloads{}
-	nottoobig, payload, _, err := CheckSizeAndSerialize(m, compress, mType)
+	notTooBigCompressed, notTooBigUncompressed, payload, _, err := CheckSizeAndSerialize(m, compress, mType)
```

Comment:
`payload` is actually the `compressedPayload`, I know its just following the existing code structure, but it would add clarity to indicate which form this is, since `CheckSizeAndSerialize` returns both.